src/transactions/mod.rs

```diff
@@ -60,6 +60,27 @@ pub const TX_VERSION: i32 = 2;
 /// <https://github.com/bitcoin/bitcoin/blob/590e49ccf2af27c6c1f1e0eb8be3a4bf4d92ce8b/src/policy/policy.h#L23-L24>
 pub const MAX_STANDARD_TX_WEIGHT: u32 = 400_000;
 
+/// This private module is used to make mutable references to the PSBT inside transaction newtypes
+/// available to functions inside the transaction module, but not beyond that. This is needed to
+/// guarantee invariants that could not be guaranteed if users had arbitrary mutable access to the
+/// inner PSBT.
+pub(super) mod inner_mut {
+    use super::{Psbt, TransactionSerialisationError};
+
+    pub trait PrivateInnerMut: Sized {
+        /// Get a mutable reference to the inner transaction, this is only used internally
+        fn psbt_mut(&mut self) -> &mut Psbt;
+
+        /// Get a reference to the inner transaction, this is only used internally
+        fn psbt(&self) -> &Psbt;
+
+        /// Move inner transaction out
+        fn into_psbt(self) -> Psbt;
+
+        fn from_psbt_serialized(raw_psbt: &[u8]) -> Result<Self, TransactionSerialisationError>;
```

Comment:
@darosior I think we can expose those three methods (at least the into_psbt and from_psbt_serialized), no?

Reply:
Yes we should actually!